src/NATS.Client.Core/NatsHeaders.cs

```diff
@@ -294,6 +294,16 @@ public void CopyTo(KeyValuePair<string, StringValues>[] array, int arrayIndex)
         }
     }
 
+    /// <summary>
+    /// Returns the bytes length of the header
+    /// </summary>
+    /// <param name="headerWriter">an instance of headerWriter</param>
+    /// <returns>bytes length of th header</returns>
+    public long GetBytesLength(HeaderWriter headerWriter)
```

Comment:
just thinking about it from the application developer standpoint: is there a way we can make this taking an optional encoding rather than the writer? maybe in that case we won't even need to expose a writer? e.g. have an internal static GrtBytesLength(Encoding) method on the writer where we can reuse here?
```csharp
public long GetBytesLength(Encoding? encoding = null)
{
    // e.g. if null set to utf-8
```
what do you think?

Reply:
@mtmk I agree, I will make some changes.